extensions/data-transfer/portability-data-transfer-mastodon/src/main/java/org/datatransferproject/transfer/mastodon/social/MastodonUtilities.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2019 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.transfer.mastodon.social;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.UrlEncodedContent;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.util.IOUtils;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import org.datatransferproject.transfer.mastodon.model.Account;
+import org.datatransferproject.transfer.mastodon.model.Status;
+
+public class MastodonUtilities {
+  private static final String ACCOUNT_VERIFICATION_URL = "/api/v1/accounts/verify_credentials";
+  private static final String STATUS_URL_PATTERN = "/api/v1/accounts/%s/statuses";
+  private static final String POST_URL = "/api/v1/statuses";
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final HttpTransport TRANSPORT = new NetHttpTransport();
+  private final URI baseUri;
+  private final String accessToken;
+  private final String baseUrl;
+
+  MastodonUtilities(String accessToken, String baseUrl) {
+    this.accessToken = accessToken;
+    this.baseUrl = baseUrl;
+    this.baseUri = URI.create(baseUrl);
```

Comment:
Could this throw an error or do we want to validate here in the constructor or a static factory method? We can also add a TODO